### PR TITLE
Add connection cache invalidation ignore logic

### LINKED ISSUE #4022: [Flaky test] dist_api_calls
### Which test is flaky?

dist_api_calls

### Since when has the test been flaky?

yes

### Link to the failed test run

https://github.com/timescale/timescaledb/runs/4927543203?check_suite_focus=true

### Log output

```bash
diff -u /home/runner/work/timescaledb/timescaledb/tsl/test/expected/dist_api_calls.out /home/runner/work/timescaledb/timescaledb/build/tsl/test/results/dist_api_calls.out
--- /home/runner/work/timescaledb/timescaledb/tsl/test/expected/dist_api_calls.out	2022-01-24 20:08:47.839263390 +0000
+++ /home/runner/work/timescaledb/timescaledb/build/tsl/test/results/dist_api_calls.out	2022-01-24 20:12:34.891341510 +0000
@@ -380,13 +380,4 @@
 (1 row)
 
 ANALYZE disttable_repl;
-SELECT approximate_row_count('disttable_repl');
- approximate_row_count 
------------------------
-                     8
-(1 row)
-
-DROP TABLE disttable_repl;
-DROP DATABASE :DN_DBNAME_1;
-DROP DATABASE :DN_DBNAME_2;
-DROP DATABASE :DN_DBNAME_3;
+ERROR:  [data_node_1]: relation "disttable_repl" does not exist
```

### Reason for flakiness

_No response_

### PR DESCRIPTION
**Pull request by pmwkaa**
Calling `ts_dist_cmd_invoke_on_data_nodes_using_search_path()` function without an active transaction allows connection invalidation event happen between applying `search_path` and the actual command execution, which leads to an error.

This change introduces a way to ignore connection cache invalidations using `remote_connection_cache_invalidation_ignore()` function.

This work is based on @nikkhils original fix and the problem research.

Fix #4022

### CODE REVIEW

File: sql/CMakeLists.txt [hunk 1]
```diff
@@ -68,6 +68,8 @@ set(TS_MODULE_PATHNAME
     ${MODULE_PATHNAME}
     PARENT_SCOPE)
 
+set(TSL_MODULE_PATHNAME "$libdir/timescaledb-tsl-${PROJECT_VERSION_MOD}" PARENT_SCOPE)
```
Comment (svenklemm) [inline sql/CMakeLists.txt:71]: That seems unrelated change.

Comment (pmwkaa) [reply sql/CMakeLists.txt:71]: It is used in the isolation test, we load different module there

**Review by akuzm — APPROVED** [review]

File: sql/CMakeLists.txt (hunk 1 above)
Comment (svenklemm) [reply sql/CMakeLists.txt:71]: this is non-TSL parts of the code so you shouldnt be referencing tsl-stuff here

Comment (pmwkaa) [reply sql/CMakeLists.txt:71]: Not sure why, this is a tsl-part isolation test. We use this approach everywhere in tsl tests to link C functions from `tsl/test/src`

**Review by nikkhils — APPROVED** [review]
LGTM
